packages/kit/src/runtime/server/page/render.js

```diff
@@ -353,17 +354,32 @@ export async function render_response({
 	// add the content after the script/css links so the link elements are parsed first
 	head += rendered.head;
 
+	const html = options.app_template({
+		head,
+		body,
+		assets,
+		nonce: /** @type {string} */ (csp.nonce)
+	});
+
 	// TODO flush chunks as early as we can
-	const html =
+	const transformed =
 		(await resolve_opts.transformPageChunk({
-			html: options.app_template({ head, body, assets, nonce: /** @type {string} */ (csp.nonce) }),
+			html,
 			done: true
 		})) || '';
 
+	if (DEV) {
+		if (transformed.split('<!--').length < html.split('<!--').length) {
+			console.warn(
+				'\u001B[1m\u001B[31mRemoving comments in transformPageChunk can break Svelte\'s hydration\u001B[39m\u001B[22m'
```

Comment:
it could be worth a comment saying what some of this stuff like `\u001B[1m` is?

Reply:
done